src/org/labkey/test/tests/assay/AssayMissingValuesTest.java

```diff
@@ -217,6 +222,54 @@ public void testAssayMV()
         testMvFiltering(List.of("age", "sex"));
     }
 
+    /**
+     * provides regression coverage for Issue 39496
+     * @throws Exception there was an exception creating the assay for this test
+     */
+    @Test
+    public void testSaveBatchAPIMissingValues() throws Exception
+    {
+        // create the assay
+        String assayName = "missingValueSaveBatchAPIAssay";
+        List<PropertyDescriptor> dataFields = List.of(
+                new FieldDefinition("ParticipantId", FieldDefinition.ColumnType.String),
+                new FieldDefinition("VisitId", FieldDefinition.ColumnType.Integer),
+                new FieldDefinition("Count", FieldDefinition.ColumnType.Integer).setMvEnabled(true));
+
+        var serverProtocol = new GeneralAssayDesign(assayName)
+                .setBatchFields(List.of(new FieldDefinition("batchData", FieldDefinition.ColumnType.String)), false)
+                .setDataFields(dataFields, false)
+                .createAssay(getProjectName(), createDefaultConnection());
+
+        @Language("JavaScript") String saveBatch = """
+                LABKEY.Experiment.saveBatch({
+                    assayId: %protocolId%,
+                    batch: {
+                        runs: [{
+                            name: 'js api',
+                            dataRows : [
+                                {participantId : 'p1', visitId : 1, count : 4.0},
+                                {participantId : 'p2', visitId : 1, count : 'N'},
+                                {participantId : 'p3', visitId : 1, count : 5, countMVIndicator : 'Q'}
+                            ]
+                        }]
+                    }
+                });
+                """.replace("%protocolId%", serverProtocol.getProtocolId().toString());
+        executeScript(saveBatch);
+
+        // navigate to the results view
+        goToProjectHome();
+        clickAndWait(Locator.linkWithText(assayName));
+        clickAndWait(Locator.linkWithText("view results"));
+        var dataRegion = DataRegionTable.DataRegion(getDriver()).waitFor();
+
+        // expect 3 rows in this assay, p2 and p3 should get mv indicators in the count column
+        Map<String, List<Integer>> expectedMVIndicators = new HashMap<>();
+        expectedMVIndicators.put("Count", List.of(1, 2));
+        checkMvIndicatorPresent(dataRegion, expectedMVIndicators);
```

Comment:
this looks good and matches the approach we discussed!
looks like the other test cases in this file also have a call to `checkDataregionData()` that goes with the `checkMvIndicatorPresent()` call.

Reply:
I've added a check for dataRegion data to the test, thanks for the suggestion